core/src/main/java/org/apache/calcite/sql/SqlOperator.java

```diff
@@ -539,8 +539,10 @@ public RelDataType inferReturnType(
       }
 
       // MEASURE wrapper should be removed, e.g. MEASURE<DOUBLE> should just be DOUBLE
-      if (isMeasure(returnType) && returnType.getMeasureElementType() != null) {
-        returnType = Objects.requireNonNull(returnType.getMeasureElementType());
+      if (opBinding.getOperator().kind != SqlKind.AS
+          && isMeasure(returnType)
+          && returnType.getMeasureElementType() != null) {
+        returnType = returnType.getMeasureElementType();
```

Comment:
The checkerframework failure below does seem related to your change

Reply:
It's related to the change you asked me to make on your code, removing the not-null assertion.

Reply:
Might need to add it back. Seems like it's messing something up.

Reply:
Also I've found it can be a time-saver to try re-building before pushing new changes.